Osb 1158 add offline mod for errands

joStrauss opened the pull request:
The discussed offline mode for errand usage.

joStrauss committed:
refactor(actions): Split functions into different files

joStrauss committed:
feat(offline): add offline mode

add documentation of the offline mode

rschollmeyer commented on actions/run.go:
```diff
@@ -0,0 +1,149 @@
+	} else {
+		// Build the given test repository as a go plugin
+		log.Printf("[INFO] Building go plugin from directory %v\n", serviceDir)
+		cmd := exec.Command("bash", "-c", "cd "+serviceDir+" && "+runtime.Version()+" build -buildmode=plugin")
```
"runtime.Version()" doesn't work here since "go1.xx.xx build"  is not a recognized command

joStrauss replied:
Fixed.

joStrauss committed:
feat(config): ca_file

Add option to add ca certificates by providing a path to a ca_file.

joStrauss committed:
feat(logger): add logger that uses stdOut and errOut

joStrauss committed:
fix(run): Use "go build" instead of "runtime.Version()"

joStrauss committed:
fix(bosh): Not starting or stopping all vms

Fixed case that providing only the instance id would stop or start ALL vms of a deployment because bosh expects {service_name}/{instance_id}.

joStrauss committed:
fix(trafficControl): not using eth0

trafficControl uses network interface with the provided IP instead of eth0 all the time.

joStrauss committed:
Merge branch 'master' into osb-1158_add_offline_mod_for_errands
